File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        current = nums[0]
        Max = nums[0]
        for i in range(1,len(nums)):
            if current+nums[i] > nums[i]:
                current=current+nums[i]
            else:
                current=nums[i]
            if current > Max:
                Max = current
        return Max
        

File: 0053-maximum-subarray/0053-maximum-subarray.c
int maxSubArray(int* nums, int numsSize) {
    int current = nums[0];
    int max = nums[0];

    for(int i = 1; i < numsSize; i++) {

        // choose current element or extend subarray
        if(current + nums[i] > nums[i])
            current = current + nums[i];
        else
            current = nums[i];

        // update maximum
        if(current > max)
            max = current;
    }

    return max;
}